### PR TITLE
Nodeinfo: We now parse Nodeinfo 2.1 and 2.2 as well

### PR DESCRIPTION
**Pull request by annando**
**Attention! Both this PR and PR #14247 are changing the database structure. Depending on which of these PRs get accepted first, the database number at the other PR has to be increased prior to accepting it.**

We have recently introduced nodeinfo in the version 2.1 and 2.2. We now parse them as well. Also previously we always fetched the Nodeinfo2 endpoint first. Since this isn't widespread in use, the request order is now reversed. The Nodeinfo2 endpoint is only queried when the request for Nodeinfo1 fails or if the fetched data is from a Nodeinfo1 1.0 endpoint.

### CODE REVIEW

File: src/Model/GServer.php [hunk 1]
```diff
@@ -87,9 +87,11 @@ class GServer
...
 	// Standardized endpoints
 	const DETECT_STATISTICS_JSON = 100;
-	const DETECT_NODEINFO_1 = 101;
-	const DETECT_NODEINFO_2 = 102;
-	const DETECT_NODEINFO_210 = 103;
+	const DETECT_NODEINFO_1 = 101; // Nodeinfo Version 1.0
+	const DETECT_NODEINFO_20 = 102; // Nodeinfo Version 2.0
+	const DETECT_NODEINFO_210 = 103; // Nodeinfo 2 Version 1.0
+	const DETECT_NODEINFO_21 = 104; // Nodeinfo Version 2.1
+	const DETECT_NODEINFO_22 = 105; // Nodeinfo Version 2.2
```
Comment (MrPetovan) [inline src/Model/GServer.php]: Renaming suggestion to thwart potential and likely confusion. We probably should do the same for the class names.
```suggestion
	const DETECT_NODEINFO1_10 = 101; // Nodeinfo (1) Version 1.0
	const DETECT_NODEINFO1_20 = 102; // Nodeinfo (1) Version 2.0
	const DETECT_NODEINFO2_10 = 103; // Nodeinfo 2 Version 1.0
	const DETECT_NODEINFO2_21 = 104; // Nodeinfo 2 Version 2.1
	const DETECT_NODEINFO2_22 = 105; // Nodeinfo 2 Version 2.2
```

File: src/Model/GServer.php [hunk 2]
```diff
@@ -1206,13 +1219,14 @@ private static function parseNodeinfo1(string $nodeinfo_url): array
...
-	 * @param string $nodeinfo_url address of the nodeinfo path
+	 * @param string $nodeinfo_url     address of the nodeinfo path
+	 * @param int    $detection_method nodeinfo version
 	 *
 	 * @return array Server data
 	 *
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 */
-	private static function parseNodeinfo2(string $nodeinfo_url): array
+	private static function parseNodeinfo2(string $nodeinfo_url, int $detection_method): array
```
Comment (MrPetovan) [inline src/Model/GServer.php]: What's the difference with `parseNodeinfo210`?

File: src/Model/GServer.php (hunk 1 above)
Comment (annando) [reply src/Model/GServer.php]: [Nodeinfo2](https://github.com/jaywink/nodeinfo2) is a fork of [Nodeinfo](https://github.com/jhass/nodeinfo). It is not version 1 or version 2.

File: src/Model/GServer.php (hunk 2 above)
Comment (annando) [reply src/Model/GServer.php]: What do you mean with that question?

Comment (MrPetovan) [reply src/Model/GServer.php]: I expect `parseNodeinfo2` to process NodeInfo2 1.0, but there's also a `parseNodeinfo210` method, so I'm wondering what's the difference between both.

**Comment by MrPetovan** [comment]
Please change the database version number now that I merged https://github.com/friendica/friendica/pull/14247

File: src/Model/GServer.php (hunk 2 above)
Comment (annando) [reply src/Model/GServer.php]: That's why I renamed the functions.

**Comment by annando** [comment]
The database version number is changed now.

**Review by MrPetovan — APPROVED** [review]

File: src/Model/GServer.php (hunk 2 above)
Comment (MrPetovan) [reply src/Model/GServer.php]: Oh, I see now, sorry for the noise.

Comment (annando) [reply src/Model/GServer.php]: No problem.

**Pull request merged**